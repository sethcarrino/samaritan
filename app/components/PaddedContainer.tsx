import React, { ReactNode } from 'react';
import { View, StyleSheet, ViewStyle } from 'react-native';

interface Props {
  children: ReactNode;
}
const PaddedContainer: React.FC<Props> = ({ children }) => (
  <View style={styles.container}>{children}</View>
);

type PaddedContainerStyleSheet = {
  container: ViewStyle;
};

const styles = StyleSheet.create<PaddedContainerStyleSheet>({
  container: {
    paddingHorizontal: 20
  }
});

export default PaddedContainer;
