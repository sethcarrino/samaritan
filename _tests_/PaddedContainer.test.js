import React from 'react';
import PaddedContainer from '../app/components/PaddedContainer';

import renderer from 'react-test-renderer';

describe('<PaddedContainer/>', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<PaddedContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
