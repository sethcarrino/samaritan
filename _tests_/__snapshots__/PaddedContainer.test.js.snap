// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaddedContainer/> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "paddingHorizontal": 20,
      },
      Object {
        "paddingTop": 0,
      },
    ]
  }
/>
`;
